tick/optim/prox/prox_group_l1.py

```diff
@@ -0,0 +1,136 @@
+    each group, or block. Blocks (non-overlapping) are specified by the
+    ``blocks_start`` and ``blocks_length`` parameters.
+
+    Parameters
+    ----------
+    strength : `float`, default=0.
+        Level of penalization. Note that `ProxL2` which is applied in each group
+        multiplies ``strength`` by the square-root of the group size.
+        This allows to consider strengths that have the same order as with
+        `ProxL1` or other separable proximal operators.
+
+    blocks_start : `list` or `numpy.array`, shape=(n_blocks,)
+        First entry of each block
+
+    blocks_length : `list` or `numpy.array`, shape=(n_blocks,)
+        Size of each block
+
+    range : `tuple` of two `int`, default=`None`
+        Range on which the prox is applied. If `None` then the prox is
+        applied on the whole vector
+
+    positive : `bool`, default=`False`
+        If True, apply the penalization together with a projection
+        onto the set of vectors with non-negative entries
+
+    Attributes
+    ----------
+    n_blocks : `int`
+        Number of blocks
+    """
+
+    _attrinfos = {
+        "strength": {
+            "writable": True,
+            "cpp_setter": "set_strength"
+        },
+        "positive": {
+            "writable": True,
+            "cpp_setter": "set_positive"
+        },
+        "blocks_start": {
+            "writable": True,
+            "cpp_setter": "set_blocks_start"
+        },
+        "blocks_length": {
+            "writable": True,
+            "cpp_setter": "set_blocks_length"
+        }
+    }
+
+    def __init__(self, strength: float, blocks_start: np.array,
+                 blocks_length: np.array, range: tuple = None,
+                 positive: bool = False):
+        Prox.__init__(self, range)
+
+        if any(length <= 0 for length in blocks_length):
+            raise ValueError("all blocks must be of positive size")
+        if any(start < 0 for start in blocks_start):
+            raise ValueError("all blocks must have positive starting indices")
+
+        if type(blocks_start) is list:
+            blocks_start = np.array(blocks_start, dtype=np.uint64)
+        if type(blocks_length) is list:
+            blocks_length = np.array(blocks_length, dtype=np.uint64)
+        if blocks_start.dtype is not np.uint64:
+            blocks_start = blocks_start.astype(np.uint64)
+        if blocks_length.dtype is not np.uint64:
+            blocks_length = blocks_length.astype(np.uint64)
+
+        if blocks_start.shape != blocks_length.shape:
+            raise ValueError("``blocks_start`` and ``blocks_length`` "
+                             "must have the same size")
+        if any(blocks_start[1:] < blocks_start[:-1]):
+            raise ValueError('``block_start`` must be sorted')
+
+        if any(blocks_start[1:] < blocks_start[:-1] + blocks_length[:-1]):
+            raise ValueError("blocks must not overlap")
```

Comment:
could'nt we mutualize this code with prox binarsity? Maybe by creating a stateless function that takes block_length and block_start as input?